### PR TITLE
Port ConsumerSharedCARingBuffer::Handle to the new IPC serialization format

### PR DESCRIPTION
**Pull request by cdumez**
#### 949236d50e8989b1aacebd090a180471ca9218e3
<pre>
Port ConsumerSharedCARingBuffer::Handle to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=265951">https://bugs.webkit.org/show_bug.cgi?id=265951</a>

Reviewed by Alex Christensen.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h:
(WebKit::ConsumerSharedCARingBufferHandle::takeOwnershipOfMemory):
(WebKit::ConsumerSharedCARingBuffer::Handle::encode): Deleted.
(WebKit::ConsumerSharedCARingBuffer::Handle::decode): Deleted.
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/271672@main">https://commits.webkit.org/271672@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/a6143c5a98457a2ca76ea8a5d3378e5454825fb8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29113 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7781 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30450 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31712 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26498 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9907 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5087 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26507 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29385 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6425 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24949 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5576 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5699 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33051 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26580 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26415 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31946 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5684 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3852 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29730 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7349 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6188 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3752 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6205 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/WebKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -18733,7 +18735,6 @@
 			isa = PBXTargetDependency;
 			platformFilters = (
 				ios,
-				xros,
```
Comment (achristensen07) [inline Source/WebKit/WebKit.xcodeproj/project.pbxproj]: This change looks unrelated.

File: Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h [hunk 2]
```diff
@@ -44,15 +44,17 @@ class SharedCARingBufferBase : public WebCore::CARingBuffer {
     Ref<SharedMemory> m_storage;
 };
 
+struct ConsumerSharedCARingBufferHandle {
+    SharedMemory::Handle memory;
+    size_t frameCount;
```
Comment (achristensen07) [inline Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h]: Default initializing to 0 couldn't hurt, and might prevent uninitialized memory.

**Review by achristensen07 — APPROVED** [review]

File: Source/WebKit/WebKit.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -18761,7 +18761,6 @@
...
```
Comment (cdumez) [inline Source/WebKit/WebKit.xcodeproj/project.pbxproj]: grrr, Xcode keeps dropping those. Very annoying.
